test/functional/ui/cmdline_spec.lua

```diff
@@ -857,4 +859,138 @@ describe("cmdline height", function()
     screen:try_resize(25, 5)
     assert_alive()
   end)
+
+  it('cmdheight can be 0', function()
```

Comment:
I'd suggest making this a `describe` block and splitting each section into a separate test (e.g.: a test for `laststatus=3`, a test for `laststatus=2`, a test for `laststatus=0`, and a test for winbar